update documentation of OPAL (#70)
bettermarks/opal: docs/auth_token_concept.md

# Authorization Concept for OPAL based on Tokens

> The OPen Adaptive Licensing (OPAL) can later be used (as a service) by providers of learning 

Reviewer: ```suggestion
The Open Adaptive Licensing (OPAL) can later be used (as a service) by providers of learning 
```

Author: showing OPAL

application (or content providers) to provide licenses for their content or (using their app) to end users - 
or groups of end users.

In a first step, OPAL should be integrated in a simple way in the "EdTech application". 
In this first step, we are not talking about fine grained permissions to be allowed to use specific parts of 
an application or specific content, we just give an 'allow all' or 'do not allow anything'.

The OPAL concept is based on the following assumptions:

> - A teacher or (school admin) or somebody else, that has an account at some IDP (we will call that IDP 'hierarchy provider') and is member of some hierarchical structures ('classes', schools', etc.) can 'purchase' a license for some 'class' or 'school' or any other structure, they are member of.

Reviewer: How about
```suggestion
- A user with an account at an IDP (referred to as the ‘hierarchy provider’) and membership in hierarchical structures (such as ‘classes’, ‘schools’, or similar entities) can purchase a license for an entity in that hierarchy they belong to.”
```

> - A student or another teacher or somebody else, that is member of a 'licensed' group or school or any other structure in the hierarchy can use that license.

Reviewer: Probably just
```suggestion
- Another member of a 'licensed' group or school or any other structure in the hierarchy can use that license.
```

In order to achieve that, we have mainly to solve these authorization problems: 

1. How can the license service know, if a teacher is allowed to purchase a license (ID=1234) for a specific class 5A?
2. How can the license service know, if a student is in group 5A and therefore may access the license (ID=1234)?

> ## Authorizing the "Classic Way"

Reviewer: shouldn't we drop this section?
